mesh/ncmesh.cpp

```diff
@@ -771,6 +809,30 @@ void NCMesh::CheckAnisoPrism(int vn1, int vn2, int vn3, int vn4,
    }
 }
 
+void NCMesh::CheckAnisoPyramid(int vn1, int vn2, int vn3, int vn4,
+                               const Refinement *refs, int nref)
+{
+   MeshId buf[4];
+   Array<MeshId> eid(buf, 4);
+   FindEdgeElements(vn1, vn2, vn3, vn4, eid);
+
+   // see if there is an element that has not been force-refined yet
+   for (int i = 0, j; i < eid.Size(); i++)
+   {
+      int elem = eid[i].element;
+      for (j = 0; j < nref; j++)
+      {
+         if (refs[j].index == elem) { break; }
+      }
+      if (j == nref) // elem not found in refs[]
+      {
+         // schedule prism refinement along Z axis
+         MFEM_ASSERT(elements[elem].Geom() == Geometry::PYRAMID, "");
+         ref_stack.Append(Refinement(elem, 4));
```

Comment:
But maybe I'm wrong and you can split a pyramid along the Z axis when a propagated aniso refinement arrives? That case should be covered by one of the unit tests then, because I'm really not sure this is correct.

Reply:
I'll give it a try and disable aniso pyramid refinement if it doesn't work. If it does work I'll leave a test in for it.

Reply:
Aniso pyramid refinement is disabled.